### PR TITLE
Streams enhancements

### PR DESCRIPTION
**Pull request by Thilas**
*** work in progress ***
I still need to test one change on my package repo.

Per #74 discussion:
- add support for common properties shared between all streams
- support alphabetical streams
- add more information on streams for plugins (+ adapt GitReleases)
- add more pester tests (+ add switch for code coverage, almost 62% in this PR)

FYI, I check that everything is fine with coreteam repo, @majkinetor repo and mine.

### CODE REVIEW

**Comment by Thilas** [comment]
I'm also creating a PR on coreteeam repo in order to use alphabetical streams on libreoffice: https://github.com/chocolatey/chocolatey-coreteampackages/pull/892.

File: AU/Private/AUPackage.ps1 [hunk 1]
```diff
@@ -14,6 +14,7 @@ class AUPackage {
 
     [string]         $StreamsPath
     [pscustomobject] $Streams
+    [hashtable]      $StreamsDetails
```
[the remark this answers is not in the document]
Comment (majkinetor) [reply AU/Private/AUPackage.ps1]: Just sayin', not that I can't live with it, given that streams are rare.

[the remark this answers is not in the document]
Comment (Thilas) [reply AU/Private/AUPackage.ps1]: Done, I've merged both properties.

[the remark this answers is not in the document]
Comment (majkinetor) [reply AU/Private/AUPackage.ps1]: Cool, feel free to merge when you are done with this PR.

File: AU/Public/Update-Package.ps1 [hunk 2]
```diff
@@ -401,30 +403,41 @@ function Update-Package {
...
+        if ($res.Streams -isnot [System.Collections.Specialized.OrderedDictionary] -and $res.Streams -isnot [HashTable]) {
+            throw "au_GetLatest doesn't return an OrderedDictionary or HashTable result for streams but $($res.Streams.GetType())"
+        }
+
+        # Streams are expected to be sorted starting with the most recent one
+        $streams = @($res.Streams.Keys)
+        # In case of HashTable (i.e. not sorted), let's sort streams alphabetically descending
+        if ($res.Streams -is [HashTable]) { $streams = $streams | sort -Descending }
 
         if ($Include) {
```
[the remark this answers is not in the document]
Comment (Thilas) [reply AU/Public/Update-Package.ps1]: Done

File: AU/Private/AUPackage.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (Thilas) [reply AU/Private/AUPackage.ps1]: Sure! I'm doing a final test and then I merge. Thanks!

**Pull request merged**